vpr/src/analytical_place/full_legalizer.cpp

```diff
@@ -281,6 +291,710 @@ static LegalizationClusterId create_new_cluster(PackMoleculeId seed_molecule_id,
+            for (auto it = tile_molecules.begin(); it != tile_molecules.end(); ) {
+                if (!cluster_legalizer.is_molecule_compatible(*it, cluster_id)) {
+                    ++it;
+                    continue;
+                }
+                if (cluster_legalizer.add_mol_to_cluster(*it, cluster_id) == e_block_pack_status::BLK_PASSED) {
+                    // If added, remove from unclustered spatial data.
+                    it = tile_molecules.erase(it);
+                } else {
+                    ++it;
+                }
+            }
+
+            // Push the neighbor tile locations onto queue (in the same layer).
+            // This will push the neighbors left, right, above, and below the current
+            // location. The code above will check if these locations are already
+            // been visited or in our search radius.
+            if (current_tile_loc.x > 0) {
+                t_physical_tile_loc new_tile_loc = t_physical_tile_loc(current_tile_loc.x - 1,
+                                                                       current_tile_loc.y,
+                                                                       current_tile_loc.layer_num);
+                loc_queue.push(new_tile_loc);
+            }
+            if (current_tile_loc.x < (int)width - 1) {
+                t_physical_tile_loc new_tile_loc = t_physical_tile_loc(current_tile_loc.x + 1,
+                                                                       current_tile_loc.y,
+                                                                       current_tile_loc.layer_num);
+                loc_queue.push(new_tile_loc);
+            }
+            if (current_tile_loc.y > 0) {
+                t_physical_tile_loc new_tile_loc = t_physical_tile_loc(current_tile_loc.x,
+                                                                       current_tile_loc.y - 1,
+                                                                       current_tile_loc.layer_num);
+                loc_queue.push(new_tile_loc);
+            }
+            if (current_tile_loc.y < (int)height - 1) {
+                t_physical_tile_loc new_tile_loc = t_physical_tile_loc(current_tile_loc.x,
+                                                                       current_tile_loc.y + 1,
+                                                                       current_tile_loc.layer_num);
+                loc_queue.push(new_tile_loc);
+            }
+        }
+        // Clean the new created cluster to avoid memory footprint increase.
+        cluster_legalizer.clean_cluster(cluster_id);
+    }
+}
+
+void FlatRecon::report_clustering_summary(ClusterLegalizer& cluster_legalizer,
+                                          std::unordered_set<PackMoleculeId>& neighbor_pass_molecules,
+                                          std::unordered_set<LegalizationClusterId>& orphan_window_clusters) {
+    // Define stat collection variables.
```

Comment:
Update comment:
Stat collection variables: key is a block_type string and value is the number of clusters of that type created.